Automattic/wp-calypso
# Domains: "Use a domain I own" flow improvements

## Proposed Changes
Depends on D108031-code (for the new endpoint).

This PR adds some small improvements to the "Use a domain I own" flow, hopefully preventing users from incorrectly using it. There are also some small CSS fixes.

## Testing Instructions
- Apply D108031-code on your back end;
- Go to the domain management page of a site you own;
- On the "Use a domain I own" flow (`Add a domain` -> `Use a domain I own`):
  - Ensure that trying to use a domain registered in WPCOM as a domain-only site displays the domain move prompt and the improved message - same as in the screenshots;
  - Ensure that trying to use a domain registered in the same site that you're currently on displays a prompt to set this domain as primary - same as in the screenshots;
  - Ensure that all other cases are unchanged.
  - Ensure that the availability request only happens if a domain is not registered with us.

## Screenshots
### Domain-only sites
#### Before
![image](https://github.com/Automattic/wp-calypso/assets/18705930/1a327b51-9338-460b-9d8e-53b7a0665d93)
#### After
![image](https://github.com/Automattic/wp-calypso/assets/18705930/72654401-13ea-46f7-8cf1-a5e3b9e56759)
### Domain registered on the same site
#### Before
![image](https://github.com/Automattic/wp-calypso/assets/18705930/123d95b8-3126-41a1-8a22-293cba7ba951)
#### After
![image](https://github.com/Automattic/wp-calypso/assets/18705930/5f7fd3b4-d577-48ab-9ccf-5e7529e85254)
### Domain registered on another site (no changes)
#### Before
![image](https://github.com/Automattic/wp-calypso/assets/18705930/f257c9c2-32d2-4efc-843f-7b053dc787f1)
#### After
![image](https://github.com/Automattic/wp-calypso/assets/18705930/bafc36fc-91c1-420e-8337-060f52cb8024)

## Pre-merge Checklist
